### ISSUE TITLE
Registering with an already-registered email causes a traceback

### ISSUE THREAD

**Issue by toolness** [GovLab/noi2#41]
Arg, internationalizing flask-security strikes again!!

Here's the traceback:

```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python2.7/dist-packages/flask_security/decorators.py", line 205, in wrapper
    return f(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/flask_security/views.py", line 116, in register
    if form.validate_on_submit():
  File "/usr/local/lib/python2.7/dist-packages/flask_wtf/form.py", line 166, in validate_on_submit
    return self.is_submitted() and self.validate()
  File "/usr/local/lib/python2.7/dist-packages/wtforms/form.py", line 310, in validate
    return super(Form, self).validate(extra)
  File "/usr/local/lib/python2.7/dist-packages/wtforms/form.py", line 152, in validate
    if not field.validate(self, extra):
  File "/usr/local/lib/python2.7/dist-packages/wtforms/fields/core.py", line 200, in validate
    stop_validation = self._run_validation_chain(form, chain)
  File "/usr/local/lib/python2.7/dist-packages/wtforms/fields/core.py", line 220, in _run_validation_chain
    validator(form, self)
  File "/usr/local/lib/python2.7/dist-packages/flask_security/forms.py", line 79, in unique_user_email
    msg = get_message('EMAIL_ALREADY_ASSOCIATED', email=field.data)[0]
  File "/usr/local/lib/python2.7/dist-packages/flask_security/utils.py", line 268, in get_message
    return rv[0] % kwargs, rv[1]
  File "/usr/local/lib/python2.7/dist-packages/speaklater.py", line 145, in __mod__
    return self.value % other
  File "/usr/local/lib/python2.7/dist-packages/speaklater.py", line 115, in <lambda>
    value = property(lambda x: x._func(*x._args, **x._kwargs))
  File "/usr/local/lib/python2.7/dist-packages/flask_babel/__init__.py", line 461, in gettext
    return t.ugettext(string) % variables
KeyError: u'email'
```

If I comment out our custom `lazy_gettext`'ed setting for `app.config['SECURITY_MSG_EMAIL_ALREADY_ASSOCIATED']` in `app/factory.py`, this traceback goes away, and the text "[email] is already associated with an account." appears as expected. So I'm guessing this was introduced by 0d52d9ae1bdc9b20acdd2c089b9c9a8a76d96492.

I suspect that the same problem might happen with `SECURITY_MSG_CONFIRM_REGISTRATION`, since it uses `%(email)s` in its gettext message too, but I haven't checked.

**Comment by talos** [comment]
@AnupamaSh Can you fix our translation strings so that the expected `%(email)s` is in there?

**Comment by toolness** [comment]
Hmm, I'm not sure if this is a problem with translation strings since I was running the site in plain english when encountering this error... I think the problem may actually have something to do with the way that the lazy string proxies are being interpolated, but I don't know enough about them to be sure... In any case, I'm willing to look into it if needed.

**Comment by talos** [comment]
Yep, @toolness is right.  @AnupamaSh if you feel stuck on this feel free to skip for now.

**Comment by toolness** [comment]
Yeah, this has to do with the way [`lazy_gettext()`](https://pythonhosted.org/Flask-Babel/#flask.ext.babel.lazy_gettext)  works... Basically, it expects to be used just like `gettext`, i.e. its signature is:

```python
def lazy_gettext(string, **variables)
```

In other words, it expects the variables (in this case, `email`) to be passed to it upon instantiation... But we're not doing that in `factory.py` (nor would it be a good idea to, since we have no conception of a "current user" in it). In any case, this results in an empty dict being passed to a string that expects the dict to contain `email`, hence the error.

The following console inspection on two different stack frames of the traceback sheds further light on the problem:

![2015-09-29_17-11-31](https://cloud.githubusercontent.com/assets/124687/10178475/ff6c8110-66cd-11e5-98c8-90216b0854f6.png)

**Comment by toolness** [comment]
Um... so one really horrible way around this is...

```python
app.config['SECURITY_MSG_EMAIL_ALREADY_ASSOCIATED'] = (
    lazy_gettext('%(email)s is already associated with an account.', email=\
'%(email)s'), 'error')
```

This will make it so that `lazy_gettext` doesn't *appear* to interpolate its argument, even though it actually does in the last stack frame, leaving the job of *actually* interpolating the string to flask-security further up the stack.

<img src="https://media.giphy.com/media/lhfnAjmZQHEpa/giphy.gif">

**Comment by talos** [comment]
Make it so.
